editor/src/core/model/element-metadata-utils.ts

```diff
@@ -2299,6 +2306,76 @@ function fillMissingDataFromAncestors(mergedMetadata: ElementInstanceMetadataMap
     }
   })
 
+  // When a conditional has no siblings, and there is a js expression in its active branch, its globalFrame/localFrame
+  // should be inherited from its parent
+  const conditionalsWithNoSiblingsAndExpressionInActiveBranch = Object.keys(workingElements).filter(
+    (p) => {
+      const element = workingElements[p]
+      if (element.conditionValue === 'not-a-conditional') {
+        return false
+      }
+      const condElement =
+        isRight(element.element) && isJSXConditionalExpression(element.element.value)
+          ? element.element.value
+          : null
+
+      if (condElement == null) {
+        return false
+      }
+
+      const activeBranch = element.conditionValue.active
+        ? condElement.whenTrue
+        : condElement.whenFalse
+
+      if (!isJSExpression(activeBranch)) {
+        return false
+      }
+
+      const parentOfConditionalPath = EP.parentPath(EP.fromString(p))
```

Comment:
```suggestion
      const parentOfConditionalPath = EP.parentPath(element.elementPath)
```
I think that should work right?